llvm/test/SYCLLowerIR/printf_addrspace/diagnose_bad_wrapper_inlining.ll

```diff
@@ -0,0 +1,47 @@
+;; This tests diagnostic emission whenever printf's format string argument
+;; is taken from the argument list of a user wrapper function, but then the
+;; user-provided wrapper does not get inlined (while the library-provided
+;; experimental::printf wrapper does).
+
+;; The IR for test purposes is based on the following source/compilation (custom
+;; build of SYCL Clang with SYCLMutatePrintfAddrspacePass turned off):
```

Comment:
Do we really need to use a custom build of clang for generation of this test?

Reply:
We do - we have to disable the printf mutation pass in `clang` before launching it via `opt`.

Reply:
Oh, otherwise we won't be able to generate this IR at all, due to the very same error, right?

Reply:
Yep. That's basically the setup that I've had to use for all of the tests - the "positive" ones were also generated this way, as otherwise the pass would've already been completed on the resulting IR.